index.html

```diff
@@ -1,18 +1,147 @@
+						>American Flatbread</a
+					>
+				</li>
+				<li>
+					<a href="http://localhost:5500/restaurant.html#august-first"
+						>August First</a
+					>
+				</li>
+				<li>
+					<a href="http://localhost:5500/restaurant.html#city-market"
+						>City Market</a
+					>
+				</li>
+				<li>
+					<a href="http://localhost:5500/restaurant.html#el-cortijo"
+						>El Cortijo</a
+					>
+				</li>
+				<li>
+					<a href="http://localhost:5500/restaurant.html#farmhouse-grill"
+						>Farmhouse Grill</a
+					>
+				</li>
+				<li>
+					<a href="http://localhost:5500/restaurant.html#hen-of-the-wood"
+						>Hen of The Wood</a
+					>
+				</li>
+				<li>
+					<a href="http://localhost:5500/restaurant.html#honey-road"
+						>Honey Road</a
+					>
+				</li>
+				<li>
+					<a href="http://localhost:5500/restaurant.html#kountry-kart-deli"
+						>Kountry Kart Deli</a
+					>
+				</li>
+				<li>
+					<a href="http://localhost:5500/restaurant.html#mr-mikes">Mr. Mikes</a>
+				</li>
+				<li>
+					<a href="http://localhost:5500/restaurant.html#pascolo-ristorante"
+						>Pascolo Ristorante</a
+					>
+				</li>
+			</ul>
+		</header>
+
+		<script>
+			// 44.4759 73.2121
+
+			var map = L.map("map").setView([44.47629, -73.213], 15);
+
+			L.tileLayer("https://{s}.tile.openstreetmap.fr/hot/{z}/{x}/{y}.png", {
+				attribution:
+					'&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+			}).addTo(map);
+
+			function getJson() {
+				fetch("/" + "all.json")
+					.then(response => {
+						return response.json();
+					})
+					.then(contents => {
+						console.log(contents);
+						contents.forEach(restaurant => {
+							fetch("/" + restaurant + ".json")
+								.then(response => {
+									return response.json();
+								})
+								.then(restaurantObject => {
+                  console.log(restaurantObject);
+									getGeoJson(restaurantObject);
+								});
+						});
+					});
+			}
+
+			function getGeoJson(location) {
+				fetch(
+					"https://nominatim.openstreetmap.org/search/?q=" +
+          encodeURIComponent(location.address) +
+						"&format=json" )
+					.then(response => {
+            return response.json();
+					})
+					.then(contents => {
+						let latitude = contents["0"].lat;
+						let longitude = contents["0"].lon;
+						//restarauntName = contents["0"].
+						//function setMarkers(latitude, longitude) {
+							L.marker([latitude, longitude])
+								.addTo(map)
+								.bindPopup(location.name);
+					});
+			}
+
+			let name = document.location.pathname.slice(1);
+
+			if (!name) {
+				console.log("no place name specfied");
+			} else {
+				console.log("fetching place named '" + name + "'");
+
+				fetch(name + ".json")
+					.then(function(response) {
+						return response.text();
+					})
+					.then(function(myText) {
+						console.log(myText);
+					});
+			}
+
+			window.addEventListener("load", getJson);
```

Comment:
This is a good use of event listeners to make sure the page gets loaded before you fetch the data to populate the page. Since your script tag is at the bottom of the page the page will load before your javascript is read so you could also just call the `getJson()` function here. Your way is safer however.